File: XTApp/XTApp/XTCommon/XTUtility/XTRoute.m
//
//  XTRoute.m
//  XTApp
//
//  Created by xia on 2024/9/5.
//

#import "XTRoute.h"
#import "XTHtmlVC.h"
#import "XTVerifyListVC.h"
#import "XTVerifyBaseVC.h"
#import "XTVerifyContactVC.h"
#import "XTOCRVC.h"
#import "XTVerifyFaceVC.h"
#import "XTVerifyBankVC.h"

@implementation XTRoute

+(instancetype)xt_share {
    static XTRoute *shareInstance;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        shareInstance = [[self alloc]init];
    });
    return shareInstance;
}

-(void)goHtml:(NSString *)url
      success:(XTBoolBlock)success {
    if(![NSString xt_isValidateUrl:url]){
        if(success){
            success(NO);
        }
        return;
    }
    XTHtmlVC *htmlVC = [[XTHtmlVC alloc] initUrl:url];
    [[XTUtility xt_getCurrentVCInNav].navigationController pushViewController:htmlVC animated:YES];
    if(success){
        success(YES);
    }
}

-(void)goVerifyList:(NSString *)productId {
    XTVerifyListVC *vc = [[XTVerifyListVC alloc] initWithProductId:productId];
    [[XTUtility xt_getCurrentVCInNav].navigationController pushViewController:vc animated:YES];
}

-(void)goVerifyItem:(NSString *)code
          productId:(NSString *)productId
            orderId:(NSString *)orderId
            success:(XTBoolBlock)success {
    if([code isEqualToString:@"AASIXTENBO"]) {
        XTVerifyBaseVC *vc = [[XTVerifyBaseVC alloc] initWithProductId:productId orderId:orderId];
        [[XTUtility xt_getCurrentVCInNav].navigationController pushViewController:vc animated:YES];
        if(success){
            success(YES);
        }
    }
    else if([code isEqualToString:@"AASIXTENBC"]) {
        XTVerifyContactVC *vc = [[XTVerifyContactVC alloc] initWithProductId:productId orderId:orderId];
        [[XTUtility xt_getCurrentVCInNav].navigationController pushViewController:vc animated:YES];
        if(success){
            success(YES);
        }
    }
    else if([code isEqualToString:@"AASIXTENBD"]) {
        XTOCRVC *vc = [[XTOCRVC alloc] initWithProductId:productId orderId:orderId];
        [[XTUtility xt_getCurrentVCInNav].navigationController pushViewController:vc animated:YES];
        if(success){
            success(YES);
        }
    }
    else if([code isEqualToString:@"AASIXTENBP"]) {
        XTVerifyFaceVC *vc = [[XTVerifyFaceVC alloc] initWithProductId:productId orderId:orderId];
        [[XTUtility xt_getCurrentVCInNav].navigationController pushViewController:vc animated:YES];
        if(success){
            success(YES);
        }
    }
    else if([code isEqualToString:@"AASIXTENBE"]) {
        XTVerifyBankVC *vc = [[XTVerifyBankVC alloc] initWithProductId:productId orderId:orderId];
        [[XTUtility xt_getCurrentVCInNav].navigationController pushViewController:vc animated:YES];
        if(success){
            success(YES);
        }
    }
    else {
        if(success){
            success(NO);
        }
    }
}

@end

File: XTApp/XTApp/XTCommon/XTUtility/XTRoute.swift
//
//  XTRoute.swift
//  XTApp
//
//  Created by Codex on 2026/4/28.
//

import UIKit

@objcMembers
@objc(XTRoute)
class XTRoute: NSObject {
    private static let shared = XTRoute()

    @objc class func xt_share() -> XTRoute {
        shared
    }

    @objc(goHtml:success:)
    func goHtml(_ url: String, success: XTBoolBlock?) {
        guard NSString.xt_isValidateUrl(url) else {
            success?(false)
            return
        }
        let htmlVC = XTHtmlVC(url: url)
        guard let navigationController = XTUtility.xt_getCurrentVCInNav()?.navigationController else {
            success?(false)
            return
        }
        navigationController.pushViewController(htmlVC, animated: true)
        success?(true)
    }

    @objc(goVerifyList:)
    func goVerifyList(_ productId: String) {
        let vc = XTVerifyListVC(productId: productId)
        XTUtility.xt_getCurrentVCInNav()?.navigationController?.pushViewController(vc, animated: true)
    }

    @objc(goVerifyItem:productId:orderId:success:)
    func goVerifyItem(_ code: String, productId: String, orderId: String, success: XTBoolBlock?) {
        let vc: UIViewController?
        switch code {
        case "AASIXTENBO":
            vc = XTVerifyBaseVC(productId: productId, orderId: orderId)
        case "AASIXTENBC":
            vc = XTVerifyContactVC(productId: productId, orderId: orderId)
        case "AASIXTENBD":
            vc = XTOCRVC(productId: productId, orderId: orderId)
        case "AASIXTENBP":
            vc = XTVerifyFaceVC(productId: productId, orderId: orderId)
        case "AASIXTENBE":
            vc = XTVerifyBankVC(productId: productId, orderId: orderId)
        default:
            vc = nil
        }

        guard let vc,
              let navigationController = XTUtility.xt_getCurrentVCInNav()?.navigationController else {
            success?(false)
            return
        }
        navigationController.pushViewController(vc, animated: true)
        success?(true)
    }
}
